### PR TITLE
Hotfix/103 customer update subscription deprecated

### PR DESCRIPTION
**Pull request by pcboy**
Fixes https://github.com/rebelidealist/stripe-ruby-mock/issues/70 and https://github.com/andrewculver/koudoku/issues/103

Related to: https://github.com/en-japan/koudoku/pull/5

### CODE REVIEW

File: app/concerns/koudoku/subscription.rb [hunk 1]
```diff
@@ -34,8 +34,10 @@ def processing!
             prepare_for_upgrade if upgrading?
 
             # update the package level with stripe.
-            customer.update_subscription(:plan => self.plan.stripe_id, :prorate => Koudoku.prorate)
-
+            if sub = customer.subscriptions.data.first
```
Comment (shawila) [inline app/concerns/koudoku/subscription.rb]: I think `first` without `data` should work

**Comment by shawila** [comment]
LGTM!
[LGTM](http://lgtm.in/i/0o01MtLWp)

[:+1:](http://lgtm.in/u/0o01MtLWp) [:-1:](http://lgtm.in/r/0o01MtLWp)

**Comment by le-doude** [comment]
LGTM

[LGTM](http://lgtm.in/i/Uk6Y6crQn)

[:+1:](http://lgtm.in/u/Uk6Y6crQn) [:-1:](http://lgtm.in/r/Uk6Y6crQn)

File: app/concerns/koudoku/subscription.rb [hunk 2]
```diff
@@ -34,8 +34,10 @@ def processing!
             prepare_for_upgrade if upgrading?
 
             # update the package level with stripe.
-            customer.update_subscription(:plan => self.plan.stripe_id, :prorate => Koudoku.prorate)
-
+            if sub = customer.subscriptions.first
+              sub.plan = self.plan.stripe_id
+              sub.save
```
Comment (mediavrog) [inline app/concerns/koudoku/subscription.rb:40]: no stripe call?

File: app/concerns/koudoku/subscription.rb [hunk 3]
```diff
@@ -86,7 +88,7 @@ def processing!
               customer = Stripe::Customer.create(customer_attributes)
 
               finalize_new_customer!(customer.id, plan.price)
-              customer.update_subscription(:plan => self.plan.stripe_id, :prorate => Koudoku.prorate)
+              customer.subscriptions.create(plan: self.plan.stripe_id, prorate: Koudoku.prorate)
```
Comment (mediavrog) [inline app/concerns/koudoku/subscription.rb:92]: stripe ruby api change?

File: app/concerns/koudoku/subscription.rb (hunk 2 above)
Comment (pcboy) [inline app/concerns/koudoku/subscription.rb:40]: This is a stripe call.

File: app/concerns/koudoku/subscription.rb (hunk 3 above)
Comment (pcboy) [inline app/concerns/koudoku/subscription.rb:92]: That's the new way to do this yeah. What do you mean?

**Comment by mediavrog** [comment]
LGTM!
[LGTM](http://lgtm.in/i/SDCLsR4cq)

[:+1:](http://lgtm.in/u/SDCLsR4cq) [:-1:](http://lgtm.in/r/SDCLsR4cq)

**Pull request merged**